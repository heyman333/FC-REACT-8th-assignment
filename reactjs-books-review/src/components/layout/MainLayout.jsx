import React from "react";
import AuthHeader from "../shared/AuthHeader";
import Footer from "../shared/Footer";
import styled from "styled-components";

const Wrap = styled.div`
  padding-top: 100px;
  display: flex;
  flex-direction: column;
`;

const Layout = styled.div.attrs({ className: "mainLayout" })`
  min-height: 500px;
`;

const MainLayout = ({ children }) => {
  return (
    <Wrap data-testid="WRAP">
      <AuthHeader />
      <Layout>{children}</Layout>
      <Footer />
    </Wrap>
  );
};

export default MainLayout;
