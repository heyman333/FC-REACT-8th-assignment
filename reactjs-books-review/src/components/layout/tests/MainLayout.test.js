import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import MainLayout from "../MainLayout";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const reducerMock = () => ({ auth: { token: "owiejfo" } });

function renderWithRedux(
  ui,
  { initialState, store = createStore(reducerMock) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

describe("<MainLayout />", () => {
  it("matches snapshot", () => {
    const utils = renderWithRedux(
      <BrowserRouter>
        <MainLayout>
          <div>MainLayout</div>
        </MainLayout>
      </BrowserRouter>
    );
    expect(utils.container).toMatchSnapshot();
  });

  it("shows header and footer when token exists", () => {
    localStorage.setItem("token", "test-token");
    const utils = renderWithRedux(
      <BrowserRouter>
        <MainLayout>
          <div>MainLayout</div>
        </MainLayout>
      </BrowserRouter>
    );
    utils.getByTestId("WRAP");
    utils.getByTestId("FOOTER");
    utils.getByTestId("HEADER");
  });
});
