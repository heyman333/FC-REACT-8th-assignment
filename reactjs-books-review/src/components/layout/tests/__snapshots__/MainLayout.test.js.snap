// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> matches snapshot 1`] = `
<div>
  <div
    class="sc-bxivhb clOneV"
    data-testid="WRAP"
  >
    <div
      class="sc-ifAKCX mainLayout cbbplG"
    >
      <div>
        MainLayout
      </div>
    </div>
    <div
      class="sc-htpNat hbjbIS"
      data-testid="FOOTER"
    />
  </div>
</div>
`;
